### ISSUE TITLE
Incorrect POST multibyte data

### ISSUE THREAD

**Issue by kosh04** [tkf/emacs-request#75]
This problem occurs when POST data includes multibyte string.

### Example

```emacs-lisp
(defun httpbin-post (data)
  (request-response-data
   (request "https://httpbin.org/post"
            :type "POST"
            :data data
            :parser #'buffer-string
            :sync t)))

(princ (httpbin-post "en=Hello&ja=こんにちは"))
{
  ...
  "form": {
    "en": "Hello", 
    "ja": "S\ufffdkao" <- garbled characters (mojibake)
  }, 
  ...
}
```

However, there is no problem if data is passed as assoc list.

```emacs-lisp
(princ (httpbin-post '(("en" . "Hello") ("ja" . "こんにちは"))))
{
  ...
  "form": {
    "en": "Hello", 
    "ja": "\u3053\u3093\u306b\u3061\u306f" <- OK
  }, 
  ...
}
```

### Workaround

Encode a data string in advance.

    (httpbin-post (encode-coding-string "en=Hello&ja=こんにちは" 'binary))

**Comment by titaniumbones** [comment]
I believe this may be a dupe of #77 , what do you think?

**Comment by titaniumbones** [comment]
Can you please test the fixes discussed in#85 and see if they work for you?

**Comment by titaniumbones** [comment]
Closing as a duplicate of #77. Please reopen if you think I'm wrong.
